go-npm-malicious-package-check/npm-malicious-package-check.go

```diff
@@ -0,0 +1,499 @@
+		if err != nil {
+			// Skip invalid patterns
+			continue
+		}
+
+		iocs = append(iocs, ioc{
+			Type:        iocType,
+			Pattern:     value,
+			Regex:       regex,
+			Description: desc,
+			Campaign:    campaign,
+		})
+	}
+
+	return iocs, nil
+}
+
+func scanPackageJSON(malicious map[string]string, path string) *finding {
+	if filepath.Base(path) != "package.json" {
+		return nil
+	}
+
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil
+	}
+
+	var pkg packageJSON
+	if err := json.Unmarshal(data, &pkg); err != nil {
+		return nil
+	}
+
+	name := strings.ToLower(strings.TrimSpace(pkg.Name))
+	version := strings.ToLower(strings.TrimSpace(pkg.Version))
+	if name == "" || version == "" {
+		return nil
+	}
+
+	pkgID := fmt.Sprintf("%s@%s", name, version)
+	context, ok := malicious[pkgID]
+	if !ok {
+		return nil
+	}
+
+	return &finding{
+		Path:    path,
+		Finding: "Malicious Package: " + pkgID,
+		Context: context,
+	}
+}
+
+func scanHostPathIOCs(iocs []ioc, path string) *finding {
+	for _, i := range iocs {
+		if i.Regex != nil && i.Regex.MatchString(path) {
+			return &finding{
+				Path:    path,
+				Finding: "IoC: " + i.Description,
+				Context: "Campaign: " + i.Campaign,
+			}
+		}
+	}
+	return nil
+}
+
+// globToRegex does a simple translation from a glob-style pattern to a regexp.
+// It supports:
+//   - ?  -> any single character
+//   - *  -> any number of non-separator characters
+//   - ** -> any number of characters, including separators
+//
+// It normalizes path separators to '/'.
+func globToRegex(pattern string) (*regexp.Regexp, error) {
+	pattern = strings.TrimSpace(pattern)
+	if pattern == "" {
+		return nil, fmt.Errorf("empty pattern")
+	}
+
+	// Expand ~ to user home where applicable
+	if strings.HasPrefix(pattern, "~") {
+		home, err := os.UserHomeDir()
+		if err == nil {
+			rest := strings.TrimPrefix(pattern, "~")
+			// Avoid double separators
+			if strings.HasPrefix(rest, "/") {
+				pattern = filepath.ToSlash(filepath.Join(home, rest[1:]))
+			} else {
+				pattern = filepath.ToSlash(filepath.Join(home, rest))
+			}
+		}
+	}
+
+	// Normalize to forward slashes
+	pattern = filepath.ToSlash(pattern)
+
+	var b strings.Builder
+	b.WriteString("^")
+
+	metaChars := `.+()|^$[]{}\-`
+
+	for i := 0; i < len(pattern); {
+		c := pattern[i]
+
+		switch c {
+		case '*':
+			// Check for **
+			if i+1 < len(pattern) && pattern[i+1] == '*' {
+				b.WriteString(".*")
+				i += 2
+			} else {
+				b.WriteString("[^/]*")
+				i++
+			}
+		case '?':
+			b.WriteString(".")
+			i++
+		default:
+			if strings.ContainsRune(metaChars, rune(c)) {
+				b.WriteByte('\\')
+			}
+			b.WriteByte(c)
+			i++
+		}
+	}
+
+	b.WriteString("$")
+	return regexp.Compile(b.String())
+}
+
+func currentUsername() string {
+	// Try common env vars first to avoid requiring os/user which may not work in all envs
+	if u := os.Getenv("USER"); u != "" {
+		return u
+	}
+	if u := os.Getenv("USERNAME"); u != "" {
+		return u
+	}
+	return "unknown"
```

Comment:
Would this work using https://pkg.go.dev/os/user#User ?

```suggestion
	if currentUser, err := user.Current(); err == nil {
	    return currentUser.Username
	} else {
	   return "unknown"
	}
```